vite-plugin/src/plugin.js

```diff
@@ -58,19 +58,26 @@ function getScssTransformsPlugin (opts) {
 }
 
 function getScriptTransformsPlugin (opts) {
+  let config
+
   return {
     name: 'vite:quasar:script',
+    configResolved(resolvedConfig) {
+      config = resolvedConfig
+    },
     transform (src, id) {
-      if (vueTransformRegex.test(id) === true) {
-        return {
-          code: vueTransform(src, opts.autoImportComponentCase),
-          map: null // provide source map if available
+      if (config.mode !== 'development') {
```

Comment:
This check here could (in theory) throw an exception if configResolved is never called, not sure how you wanted to handle that.